### PR TITLE
Remove unused files in sql and extensions folders

### PR DESCRIPTION
**Pull request by corivera**
I was looking around in the modal dialog code when I noticed that calloutDialog.ts wasn't being referenced anywhere. I then ran through the rest of the repo with the ts-unused-exports tool to find any others which could be removed. These were the ones I found outside of the vs folder that didn't look like they were in any prototype stages (like ExtHostPerf or SqlNotebookController). Let me know if there are any ones we should keep.

I also tried checking for unused exports, but the tool was generally hitting things like types that were used outside a file but not explicitly referenced by name. Removing export for those seemed like it would be confusing, even though it would still build fine.

### CODE REVIEW

**Review by llali — APPROVED** [review]

File: extensions/github-authentication/src/env/browser/authServer.ts
Comment (kburtram) [inline extensions/github-authentication/src/env/browser/authServer.ts:6]: we should probably leave the extension we've forked from vscode as-is.

Comment (corivera) [reply extensions/github-authentication/src/env/browser/authServer.ts]: Restored those files.

**Review by kburtram — APPROVED** [review]
Overall looks good to me, but please get sign-off from the various component committers before merging since this appears to be making updates broadly across the code base.  From what I can tell this looks like unused or abandoned code.

File: extensions/arc/src/test/mocks/fakeControllerModel.ts
Comment (Charles-Gagnon) [inline extensions/arc/src/test/mocks/fakeControllerModel.ts:1]: @candiceye Instead of deleting the ones in arc/azcli let's just comment them out - they were used by tests that were commented out a long time back but unless those are actually deleted it'd be preferable to do the same here so that when they're (hopefully) re-enabled the person doing that won't have to dig through a ton of history to get these back.

@candiceye FYI, although I suppose if you don't think you'll ever invest in re-enabling these tests we can just delete all the test stuff from these extensions completely.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**
